### PR TITLE
🚨 Improve onboarding error handling 

### LINKED ISSUE #818: Display email errors on login screen
## Description
Display email errors on login screen (as we do in the signup one)

## Definition of done

- [ ] invalid emails produce an error displayed below the email field

- [ ] The documentation is updated to reflect the changes done in this PR

### LINKED ISSUE #819: Display error when google login fails
## Definition of done

- [ ] when google login fails, an error toast is shown

### LINKED ISSUE #820: Clear user errors when navigating
## Definition of done

- [ ] when error happens on sso screen, it should not carry over to the email login screen
- [ ] when error happens during google login it should not carry over to sso and email login screens
- [ ] when error happens during regular signup/login it is not carried over to sso screen
- [ ] scenarios not described above are indentified and handled too

- [ ] The documentation is updated to reflect the changes done in this PR

### PR DESCRIPTION
**Pull request by pdrj**
### Summary
- Clears errors when moving between various onboarding screens.
- Displays google login errors via toast. 
- Displays email validation errors on login screen.

### Technical considerations
I introduced another common reducer to handle the cleanup on back presses.

### Relationships

Closes #819 
Closes #820 
Closes #818 

### Pull Request checklist

- [x] The code does not hinder the app's performance
- [x] The code does not introduce regressions
- [x] The code in this PR is tested thoroughly enough that changing the behavior to an incorrect one breaks tests
- [ ] One of our [live or file custom templates](https://github.com/toggl/android/tree/master/.idea/templates) was used in this PR

### Pointers
Trigger errors in various screens and validate you can't see them in other screens. Trigger a google error and validate you see a toast.

### Review template

- [ ] The code is thoroughly tested in well written and resilient manner
- [ ] The code is performant and leads to an optimal user experience
- [ ] The code is testable and does not cut corners for the sake of delivering
- [ ] The code considers older platforms well and will work in older devices just like it will in new ones

- [ ] The code considers light and dark theme
- [ ] The code considers RTL and does not mention left/right
- [ ] The code considers i18n
  - [ ] There are no hardcoded strings in the UI
  - [ ] All strings are included in every language we support OR there's an issue requesting new translations
- [ ] There are no hard-coded numbers referring to dimensions or font sizes
- [ ] The code considers UI elements that are similar and extracts their similarities into styles to avoid duplication

## Merge permissions
Anyone

### CODE REVIEW

File: onboarding/src/main/java/com/toggl/onboarding/welcome/ui/WelcomeFragment.kt [hunk 1]
```diff
@@ -93,9 +97,14 @@ class WelcomeFragment : Fragment(R.layout.fragment_welcome) {
...
         store.state
             .map { it.termsAccepted }
-            .filter { it == true }
+            .filter { it }
             .onEach { store.dispatch(WelcomeAction.SignUserUpWithThirdParty) }
             .launchIn(lifecycleScope)
+
+        store.state
+            .mapNotNull { if (it.backStack.last() is Route.Welcome) it.user as? Loadable.Error else null }
+            .onEach { context?.run { Toast.makeText(this, it.failure.errorMessage, Toast.LENGTH_LONG).show() } }
```
Comment (daividssilverio) [inline onboarding/src/main/java/com/toggl/onboarding/welcome/ui/WelcomeFragment.kt:106]: The only problem I see with this approach is that if the state somehow changes while the user error is still in the state, the toast will be duplicated. But it works for now.

**Review by daividssilverio — APPROVED** [review]
Minor concern about duplicated toasts, but LGTM.

File: onboarding/src/main/java/com/toggl/onboarding/welcome/ui/WelcomeFragment.kt (hunk 1 above)
Comment (pdrj) [reply onboarding/src/main/java/com/toggl/onboarding/welcome/ui/WelcomeFragment.kt:106]: That's a good point. For this to happen though, first we'd have to have an error during google login and the second attempt would have to go through an action that modifies state and not the user, so for instance the 'must accept terms' one. Then, when we're back to the welcome screen, a toast should appear again. 

I'll add a cleanup step that happens each time the google button is pressed.

**Comment by pdrj** [comment]
I've set the user to loading on each google login attempt @daividssilverio. Is there any other scenario you can think of that would trigger the double toast?

**Comment by daividssilverio** [comment]
> Is there any other scenario you can think of that would trigger the double toast?

Not really @pdrj. :shipit:

**Pull request merged**